mne/beamformer/tests/test_dics.py

```diff
@@ -570,6 +575,69 @@ def test_apply_dics_timeseries(_load_forward, idx):
         apply_dics_epochs(epochs, filters_vol)
 
 
+@testing.requires_testing_data
+@pytest.mark.parametrize('return_generator', (True, False))
+def test_apply_dics_tfr(return_generator):
+    """Test lcmv applied to time-frequency objects."""
+    rng = np.random.default_rng(11)
+    n_epochs = 8
+    info = read_info(fname_raw)
+    info = pick_info(info, pick_types(info, meg='grad'))
+    forward = mne.read_forward_solution(fname_fwd)
+    freqs = np.arange(8, 10)
+    times = np.linspace(0, 1, int(round(info['sfreq'])))
+    data = rng.random((n_epochs, len(info.ch_names), freqs.size, times.size))
+    data *= 1e-6
+    epochs_tfr = EpochsTFR(info, data, times=times, freqs=freqs)
+    epochs_tfr.apply_baseline((0, 0.5))
+    csd = csd_tfr(epochs_tfr, tmin=0.5, tmax=1)
+    pick_ori = None  # csd doesn't support vector so can't use for testing
+    filters = make_dics(epochs_tfr.info, forward, csd,
+                        pick_ori=pick_ori, reg=0.05,
+                        real_filter=False)
+    with pytest.raises(RuntimeError,
```

Comment:
This whole test function could use some more comments on what you are testing where (i.e., real data, complex data etc). Will make it easier to trouble shoot in the future if necessary.